### PR TITLE
[api] Use in? instead of include? where it makes sense

### PR DESCRIPTION
**Pull request by bgeuken**
eg. my_var.in?(["a", "b", "c"]) vs ["a", "b", "c"].include?(my_var)

The latter versions puts the focus on ["a", "b", "c"], while we actually
care about the content of my_var.

http://api.rubyonrails.org/classes/Object.html#method-i-in-3F

### CODE REVIEW

File: src/api/app/controllers/source_controller.rb [hunk 1]
```diff
@@ -173,15 +173,15 @@ def project_command
...
+    if params[:cmd] && !params[:cmd].in?(valid_commands)
       raise IllegalRequest, 'invalid_command'
     end
 
     command = params[:cmd]
     project_name = params[:project]
     params[:user] = User.current.login
 
-    if %w(undelete release copy move).include?(command)
+    if command.in?("undelete", "release", "copy", "move")
```
Comment (Ana06) [inline src/api/app/controllers/source_controller.rb]: that doesn't work, `in?` only accept one parameter, so it should be:

``` Ruby
command.in?(["undelete", "release", "copy", "move"])
```

or

``` Ruby
command.in? %w(undelete release copy move)
```

File: src/api/app/controllers/source_controller.rb [hunk 2]
```diff
@@ -291,7 +291,7 @@ def require_package
     @deleted_package = params.has_key? :deleted
 
     # FIXME: for OBS 3, api of branch and copy calls have target and source in the opossite place
-    if ['branch', 'release'].include? params[:cmd]
+    if params[:cmd].in?('branch', 'release')
```
Comment (Ana06) [inline src/api/app/controllers/source_controller.rb]: same here :confounded:

File: src/api/app/controllers/source_controller.rb [hunk 3]
```diff
@@ -351,11 +351,9 @@ def package_command
     # Check for existens/access of origin package when specified
     @spkg = nil
     Project.get_by_name origin_project_name if origin_project_name
-    # rubocop:disable Metrics/LineLength
-    if origin_package_name && !%w(_project _pattern).include?(origin_package_name) && !(params[:missingok] && ['branch', 'release'].include?(@command))
+    if origin_package_name && !origin_package_name.in?("_project", "_pattern") && !(params[:missingok] && @command.in?('branch', 'release'))
```
Comment (Ana06) [inline src/api/app/controllers/source_controller.rb]: and here :confounded:

File: src/api/app/controllers/source_controller.rb [hunk 4]
```diff
@@ -351,11 +351,9 @@ def package_command
...
       @spkg = Package.get_by_project_and_name(origin_project_name, origin_package_name)
     end
-    # rubocop:enable Metrics/LineLength
```
Comment (Ana06) [inline src/api/app/controllers/source_controller.rb]: unrelated to this PR, right? I would put it in a separate commit

File: src/api/app/helpers/webui/buildresult_helper.rb [hunk 5]
```diff
@@ -14,7 +14,7 @@ def arch_repo_table_cell(repo, arch, package_name, status = nil, enable_help = t
     end
 
     result = content_tag(:td, class: [theclass, "buildstatus", "nowrap"]) do
-      if %w(- unresolvable blocked excluded scheduled).include?(code)
+      if code.in?("-", "unresolvable", "blocked", "excluded", "scheduled")
```
Comment (Ana06) [inline src/api/app/helpers/webui/buildresult_helper.rb]: and here :confounded: :confounded:

File: src/api/app/models/user.rb [hunk 6]
```diff
@@ -332,7 +332,7 @@ def self.realname_for_login(login)
   # state an password hash type being in the range of allowed values.
   def validate
     # validate state and password has type to be in the valid range of values
-    errors.add(:password_hash_type, 'must be in the list of hash types.') unless User.password_hash_types.include? password_hash_type
+    errors.add(:password_hash_type, 'must be in the list of hash types.') unless password_hash_type.in?(User.password_hash_types.include?)
```
Comment (Ana06) [inline src/api/app/models/user.rb]: `password_hash_type.in?(User.password_hash_types)` you left the `include?` :rofl:

File: src/api/app/controllers/source_controller.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/app/controllers/source_controller.rb]: Yep, already noticed. I'll change it later.

**Comment by Ana06** [comment]
> [api] Use in? instead of include? where it makes sense, part II

part II? :rofl: Is there any reason to not squash the commits? :confused:

**Comment by bgeuken** [comment]
@Ana06 haha, is that payback for [this](https://github.com/openSUSE/open-build-service/pull/2687#issuecomment-280013927)? :-)

Only reasoning for splitting them was the amount of changes. So if I would have broken some tests, I would have an easier time finding the faulty change.

I'll squash them now.
